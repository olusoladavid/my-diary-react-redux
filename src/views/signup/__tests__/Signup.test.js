import thunk from "redux-thunk";
import configureStore from "redux-mock-store";

import ConnectedSignup, { Signup } from "../Signup";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const storeState = {
  user: { isAuthenticated: false },
  loading: false
};

const props1 = {
  hideToast: jest.fn(),
  createUser: jest.fn(),
  showToast: jest.fn(),
  loading: false,
  isAuthenticated: false
};

const props2 = {
  ...props1,
  loading: true,
  isAuthenticated: false
};

const props3 = {
  ...props1,
  isAuthenticated: true
};

it("renders without crashing", () => {
  const wrapper = shallow(<Signup />);
  expect(wrapper).toMatchSnapshot();
});

it("renders without crashing when user is authenticated", () => {
  const wrapper = shallow(<Signup {...props3} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders without crashing when user is signing up", () => {
  const wrapper = shallow(<Signup {...props2} />);
  expect(wrapper).toMatchSnapshot();
});

it("clears errors when input fields are focused", () => {
  const wrapper = shallow(<Signup {...props1} />);
  wrapper
    .find(".form__input")
    .at(0)
    .simulate("focus");
  expect(props1.hideToast).toHaveBeenCalled();
});

it("should save user input tp state", () => {
  const wrapper = shallow(<Signup {...props1} />);
  wrapper
    .find(".js-signup-email")
    .at(0)
    .simulate("change", { target: { value: "email@email.com", name: "email" } });
  expect(wrapper.state("email")).toEqual("email@email.com");
});

it("should not submit form when passwords differ", () => {
  const wrapper = shallow(<Signup {...props1} />);
  wrapper.setState({ email: "email", password: "password", repassword: "password1" });
  wrapper
    .find(".js-form-signup")
    .at(0)
    .simulate("submit", { preventDefault: jest.fn() });
  expect(props1.createUser).not.toHaveBeenCalled();
});

it("should submit form when passwords are the same", () => {
  const wrapper = shallow(<Signup {...props1} />);
  wrapper.setState({ email: "email", password: "password", repassword: "password" });
  wrapper
    .find(".js-form-signup")
    .at(0)
    .simulate("submit", { preventDefault: jest.fn() });
  expect(props1.createUser).toHaveBeenCalled();
});

it("should synchronise with app state in the store", () => {
  const store = mockStore(storeState);
  const wrapper = shallow(<ConnectedSignup {...props3} store={store} />);
  expect(wrapper.prop("isAuthenticated")).toBe(false);
});
