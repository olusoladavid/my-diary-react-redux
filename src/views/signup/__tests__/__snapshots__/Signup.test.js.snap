// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Fragment>
  <div
    className="container"
  >
    <div
      className="form-group"
    >
      <h2
        className="form-group__heading"
      >
        Create an account
      </h2>
      <form
        action="#"
        className="form-group__form js-form-signup"
        onSubmit={[Function]}
      >
        <input
          className="form__input js-signup-email"
          name="email"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="email"
          required={true}
          type="email"
          value=""
        />
        <input
          className="form__input js-signup-password"
          minLength={5}
          name="password"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="password"
          required={true}
          type="password"
          value=""
        />
        <input
          className="form__input js-signup-repassword"
          minLength={5}
          name="repassword"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="confirm password"
          required={true}
          type="password"
          value=""
        />
        <button
          className="form__btn--submit js-login-button"
          type="submit"
        >
          Signup
        </button>
        <Link
          className="form__btn--alternate"
          replace={false}
          to="/login"
        >
          Login?
        </Link>
        <p
          className="form__error js-error-field"
        />
      </form>
    </div>
  </div>
</Fragment>
`;

exports[`renders without crashing when user is authenticated 1`] = `
<Redirect
  push={false}
  to="/"
/>
`;

exports[`renders without crashing when user is signing up 1`] = `
<Fragment>
  <div
    className="container"
  >
    <div
      className="form-group"
    >
      <h2
        className="form-group__heading"
      >
        Create an account
      </h2>
      <form
        action="#"
        className="form-group__form js-form-signup"
        onSubmit={[Function]}
      >
        <input
          className="form__input js-signup-email"
          name="email"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="email"
          required={true}
          type="email"
          value=""
        />
        <input
          className="form__input js-signup-password"
          minLength={5}
          name="password"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="password"
          required={true}
          type="password"
          value=""
        />
        <input
          className="form__input js-signup-repassword"
          minLength={5}
          name="repassword"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="confirm password"
          required={true}
          type="password"
          value=""
        />
        <button
          className="form__btn--submit js-login-button"
          type="submit"
        >
          <i
            className="fas fa-circle-notch fa-spin"
          />
        </button>
        <Link
          className="form__btn--alternate"
          replace={false}
          to="/login"
        >
          Login?
        </Link>
        <p
          className="form__error js-error-field"
        />
      </form>
    </div>
  </div>
</Fragment>
`;
